### PR TITLE
Add magazine sensor edge reporting

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
This replaces the magazine loss/lash reporting (edge reports will allow
this to be handled on the host instead)

### CODE REVIEW

**Comment by Jeff-Ciesielski** [comment]
FWIW: Nothing in production made use of the slop reports, they were only used for doing  some analysis on how much braking would reduce lost counts.  Taking this out doesn't affect anything upstream.

File: magazine.c [hunk 1]
```diff
@@ -23,123 +25,67 @@ struct probe_window {
...
+  mag_state.on_probe = magazine_alignment_on;
 }
 
 // Monitors the gap in units between mags and throws an alarm if the gap is larger than a
 // specified threshold. Additionally, magazine slop can be calculated
 // if turned on via compiler flag
 void magazine_gap_monitor()
 {
-  const uint8_t magazine_alignment_on = magazine_get_state();
+  const bool magazine_alignment_on = !!magazine_get_state();
```
Comment (StephanCoetzee) [inline magazine.c]: magazine_get_state already returns a boolean, so the !! is not necessary.

File: progman.c [hunk 2]
```diff
@@ -49,8 +49,8 @@ bool progman_handle_runtime_cmd(char data)
   case CMD_LIMIT_REPORT:
     request_report(REQUEST_LIMIT_REPORT,0);
     break;
-  case CMD_SLOP_REPORT:
-    request_report(REQUEST_SLOP_REPORT,0);
+  case CMD_EDGE_REPORT:
+    request_report(REQUEST_EDGE_REPORT,0);
```
Comment (StephanCoetzee) [inline progman.c:53]: whitespace around comma

**Comment by StephanCoetzee** [comment]
Other than the two minor things that I mentioned, it LGTM assuming you tested it.

File: magazine.c [hunk 3]
```diff
@@ -23,123 +25,67 @@ struct probe_window {
...
-
-    magazine_calculate_loss();
-    request_report(REQUEST_SLOP_REPORT, 0);
+  // We shouldn't send reports directly from here (as this is in an
+  // interrupt context).  Instead, we queue them up and will send them
+  // in the main program loop via a report request
+  if (magazine_alignment_on != mag_state.on_probe) {
+    struct edge_event evt = {0};
+    evt.state = magazine_alignment_on;
+    evt.position = sys.position[C_AXIS];
```
Comment (StephanCoetzee) [inline magazine.c:66]: There is no way of knowing on which axis the event took place by looking at the struct. Is this a problem? I know for now we're only doing this test on the C_AXIS. Do we want to generalize edge events to all axes in the future?

File: magazine.c [hunk 4]
```diff
@@ -23,123 +25,67 @@ struct probe_window {
...
-  } else if (mag_state.on_probe) {
-    mag_state.on_probe = false;
-    magazine_record_edge(E_MAGAZINE_EDGE_TYPE_FALLING);
-    mag_state.evt_no = mag_state.evt_no ? 0 : 1;
-
-    magazine_calculate_loss();
-    request_report(REQUEST_SLOP_REPORT, 0);
+  // We shouldn't send reports directly from here (as this is in an
+  // interrupt context).  Instead, we queue them up and will send them
+  // in the main program loop via a report request
```
Comment (StephanCoetzee) [inline magazine.c:62]: :+1:

File: magazine.c (hunk 3 above)
Comment (Jeff-Ciesielski) [reply magazine.c:66]: I think this one is pretty specialized and will always be on this axis.  In the future, we'll have to think about generalizing.

File: progman.c (hunk 2 above)
Comment (Jeff-Ciesielski) [reply progman.c:53]: I'm matching surrounding code.  I agree that it is ugly.

**Comment by Jeff-Ciesielski** [comment]
Updated

**Comment by StephanCoetzee** [comment]
LGTM

**Pull request merged**